src/Admin/Extension/CreateSnapshotAdminExtension.php

```diff
@@ -50,6 +50,10 @@ private function createByPage(object $object): void
     {
         if ($object instanceof PageBlockInterface) {
```

Comment:
```
if ($object instanceof PageBlockInterface) {
    $object = $object->getPage();
}

if (!$object instanceof PageInterface) {
    return;
}

Reply:
I don't like a lot to replace variables that comes from parameters, but I will try to find a cleaner option.

Reply:
I didn't found a cleaner solution, I applied your patch.